llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -549,33 +549,37 @@ static TableFiles processOneModule(std::unique_ptr<Module> M, bool IsEsimd,
   bool SpecConstsMet = false;
   bool SetSpecConstAtRT = DoSpecConst && (SpecConstLower == SC_USE_RT_VAL);
 
+  if (DoSplit) {
+    splitModule(*M, GlobalsSet, ResultModules);
+    // post-link always produces a code result, even if it is unmodified input
+    if (ResultModules.size() == 0)
```

Comment:
```suggestion
    if (ResultModules.empty())
```

Reply:
Fixed in 3c7608bb910dcc923c2a1a6401404435a0976264